force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,46 @@
+public class OpportunityTriggerHandler {
+    public static void beforeUpdate(List<Opportunity> oppList) {
+        for (Opportunity opp : oppList) {
+            if (opp.Amount <= 5000) {
+                opp.addError('Opportunity amount must be greater than 5000');
+            }
+        }
+        Set<Id> accountIds = new Set<Id>();
+        
+        for (Opportunity opp : oppList) {
```

Comment:
Consolidate the loops in this handler as well or replace them with calls to separate methods. Its okay to pass the oppList to multiple methods that each have their own responsibility/funcitonality.